app/Tasks/Start.php

```diff
@@ -28,7 +28,10 @@ public function run(...$args): int
 
         if ($this->checks) {
             array_unshift($tasks, [$this, 'checkDependencies']);
-            $tasks[] = [$this, 'checkDatabase'];
+
+            if ($this->checkDatabaseAvailability) {
+                $tasks[] = [$this, 'checkDatabase'];
+            }
```

Comment:
I think that this check should be outside of `$this->checks`, so they are isolated